apps/jobs/views.py

```diff
@@ -336,6 +336,10 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
                     if submission_serializer.is_valid():
                         submission_serializer.save()
 
+        # Override submission visibility if leaderboard_public = False for a challenge phase
+        if not challenge_phase.leaderboard_public:
+            request.data["is_public"] = challenge_phase.leaderboard_public
```

Comment:
It should be `challenge_phase.leaderboard_public` -> `challenge_phase.is_submission_public`, no?

Reply:
is it possible for hosts to set `leaderboard_public=False` and `is_submission_public=True` ? both will be same right?

Reply:
Yes, it is possible for them to set these things and they both are different fields.